core/src/main/java/org/elasticsearch/search/aggregations/AbstractAggregationBuilder.java

```diff
@@ -116,6 +116,11 @@ public AB setMetaData(Map<String, Object> metaData) {
     }
 
     @Override
+    public Map<String, Object> getMetaData() {
```

Comment:
If the user is attempting to add metadata to a request, and doesn't have all the metadata in one place in their application, it wouldn't be uncommon to do something like:

```java
Map<String, Object> newMetadata = new HashMap<>();
newMetadata.putAll(aggBuilder.getMetadata());
newMetadata.put("foo", "bar");
aggBuilder.setMetadata(newMetadata);
```

E.g. without a way to get metadata, the metadata has to be compiled into a single location in the user's application so it can be set all at once, which makes building agg trees cumbersome.
newMetadata.

Also, being able to get the metadata would is useful where aggs are built programmatically and you want to verify the final agg builder in a test.